### ISSUE TITLE
Adapt to object detection problems

### ISSUE THREAD

**Issue by RizhaoCai** [google-research/morph-net#41]
Hello. Your work is great.

May I ask for your advice for adapting it to object detection problems?

**Comment by eladeban** [comment]
Hi,

For the network stem you can use MorphNet directly, you will be able to potentially speed up the creation of the embedding. 

For the region proposal part of the network, you could derive similar sparseing regularization terms that could trade-off compute and accuracy.

**Comment by athorneak13** [comment]
Hi eladeban, 

Wanted to see if you could elaborate on your prior comment? From the research that I've done, it seems that most ssd object detection algorithms are outputting weights for regression (bounding box predictions) and class prediction at the same time. Which embeddings are you talking about? 

Also, does anyone know if there is a way to have access to the object detection (https://github.com/tensorflow/models/tree/master/research/object_detection) model architecture to understand better how to use morphnet on it? We are trying to use the base code given to us, and we don't have labels to pass into the morphnet model and we aren't sure where we can find them/what format they should be in. 

Thanks!

**Comment by eladeban** [comment]
I think of the embedding as the output of the 'stem' part of the network, that is the part the is taken from image classification networks such as inception/resnet/mobilenet.

MorphNet dose not require 'labels' it only needs the op (accessed by `tensor.op`) which is the start to start the graph traversal at. It could be logits.op in a classification model other ops in different networks.
